app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/viewmodel/VideosInSameCategoryViewModel.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package android.support.v17.leanback.supportleanbackshowcase.app.room.viewmodel;
+
+import android.app.Application;
+import android.arch.core.util.Function;
+import android.arch.lifecycle.AndroidViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MediatorLiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.ViewModelProvider;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.config.AppConfiguration;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.DatabaseHelper;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.entity.VideoEntity;
+
+import java.util.List;
+
+public class VideosInSameCategoryViewModel extends AndroidViewModel {
+
+    private static final MutableLiveData ABSENT = new MutableLiveData();
+
+    // The parameter used to create view model
+    private final String mCategory;
+
+    private DatabaseHelper mDatabaseHelper;
+
+    /**
+     * List of VideoEntities in same category
+     */
+    private final LiveData<List<VideoEntity>> mVideosInSameCategory;
+
+    {
+        ABSENT.setValue(null);
+    }
+
+    public VideosInSameCategoryViewModel(@NonNull Application application, final String category) {
+        super(application);
+        this.mCategory = category;
+
+        mDatabaseHelper = DatabaseHelper.getInstance();
+
+        final LiveData<Boolean> databaseUpdated = mDatabaseHelper.getDatabaseUpdatedSignal();
+
+        mVideosInSameCategory = Transformations.switchMap(mDatabaseHelper.isDatabaseCreated(),
+                new Function<Boolean, LiveData<List<VideoEntity>>>() {
+                    @Override
+                    public LiveData<List<VideoEntity>> apply(Boolean isDatabaseCreated) {
+
+                        // If the database has not been created, return a null value wrapped in the
+                        // live data
+                        if (!isDatabaseCreated) {
+                            return ABSENT;
+                        }
+
+                        return Transformations.switchMap(databaseUpdated,
+
```

Comment:
databaseUpdated is totally unnecessary. Just return a live data from dao.

Reply:
LiveData is awesome! It is really good that this functionality has already been integrated into live data